src/e2e-test/java/io/cdap/cdap/ui/stepsdesign/BeforeActions.java

```diff
@@ -36,13 +37,23 @@ public void loginIfRequired() throws IOException {
     Helper.loginIfRequired();
   }
 
-  @Before(order = 1, value = "@SOURCE_CONTROL_MANAGEMENT_TEST")
-  public void setGitBranchConfig() {
+  @Before(order = 1, value = "@SOURCE_CONTROL_MANAGEMENT_CONFIG_CRUD_TEST")
+  public void setGitRepoConfig() {
+    addRepoCredentials();
+  }
+
+  @Before(order = 1, value = "@SOURCE_CONTROL_MANAGEMENT_SYNC_APPS_TEST")
+  public void setGitBranchConfig() throws GitAPIException, IOException {
+    addRepoCredentials();
     if (Strings.isNullOrEmpty(PluginPropertyUtils.pluginProp(Constants.GIT_BRANCH_PROP_NAME))) {
```

Comment:
Is this not needed in SOURCE_CONTROL_MANAGEMENT_CONFIG_CRUD_TEST also ?
We can make `setGitBranchConfig` -> `createGitBranchConfig`

Reply:
hmmm, good point, actually we can add the test step for the branch as well

Reply:
Then I don't need to have two suite test tags. Just have one `SOURCE_CONTROL_MANAGEMENT_TEST` and create and delete the branch in all SCM related tests.